### DISCUSSION TITLE
Keep current (but unchecked) streak at top of widget until day ends

### DISCUSSION THREAD

**Discussion by abathur** [iSoron/uhabits#278] [Feature Requests]
Apologies if this is a known issue; I tried a few searches but didn't see anything obvious.

I have a widget showing 9 streaks for a habit where all shown streaks are now over two weeks. My current streak will show at the top of the widget even when it's shorter than this--but only if I've already checked off the habit for the day (I prefer to check it off at bedtime).

This means that, for the first ~2 weeks of a new streak, I have to prematurely check it off if I want to be able to see my current streak before bedtime.

I see two potential solutions:

1. Don't count the "pending" streak as broken until the day has ended.
2. Always show the most-recent streak atop the widget (the pending streak would remain visible, regardless of length, until you mark the habit to begin a new streak).

In either case, it probably makes sense to indicate a "pending" streak line with an outline, background, or distinct text color.

**Comment by iSoron** [comment]
Thank you for the suggestion, @abathur. I think solution 2 is acceptable. We can simply show the most recent streak at the top.
